sway/input/text_input.c

```diff
@@ -431,18 +436,37 @@ static void handle_im_popup_destroy(struct wl_listener *listener, void *data) {
 	struct sway_input_popup *popup =
 		wl_container_of(listener, popup, popup_destroy);
 	wl_list_remove(&popup->focused_surface_unmap.link);
-	wl_list_remove(&popup->popup_surface_commit.link);
+	wl_list_remove(&popup->popup_surface_map.link);
+	wl_list_remove(&popup->popup_surface_unmap.link);
 	wl_list_remove(&popup->popup_destroy.link);
 	wl_list_remove(&popup->link);
 
 	free(popup);
 }
 
-static void handle_im_popup_surface_commit(struct wl_listener *listener,
-		void *data) {
+static void handle_im_popup_surface_map(struct wl_listener *listener, void *data) {
 	struct sway_input_popup *popup =
-		wl_container_of(listener, popup, popup_surface_commit);
-	input_popup_update(popup);
+		wl_container_of(listener, popup, popup_surface_map);
+	popup->scene_tree = wlr_scene_subsurface_tree_create(root->layers.popup,
+		popup->popup_surface->surface);
```

Comment:
Hm, would it be simpler to toggle the scene's node `enabled` based on the `mapped` flag?

(Doesn't the scene surface tree already do this?)